packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -165,7 +167,12 @@ export class TeamsAppSolution implements Solution {
     return ok({});
   }
 
-  fillInSolutionSettings(ctx: SolutionContext): Result<AzureSolutionSettings, FxError> {
+  assertContextNotEmpty(
+    ctx: SolutionContext
+  ): Result<
+    { answers: Inputs; projectSettings: ProjectSettings; solutionSettings: SolutionSettings },
```

Comment:
To keep consistent with the input parameter and method name, why not simply return the `SolutionContext` instead of another object? All these property won't be modified and still can be accessed from `SolutionContext`.

Reply:
Just do not want to use the non-null assertion operator. Split this function to three smaller assert functions.

Reply:
Cool.